samples/rust/rust_selftests.rs

```diff
@@ -71,12 +71,44 @@ macro_rules! do_tests {
     }
 }
 
-/// An example of test.
-#[allow(dead_code)]
-fn test_example() -> Result<TestSummary> {
-    // `use` declarations for the test can be put here, e.g. `use foo::bar;`.
+fn test_rust_smp_cpu() -> Result<TestSummary> {
+    use kernel::cpu::Cpu;
+
+    if Cpu::num_possible() == 1 {
+        // Nothing more to do on a single-CPU system.
+        pr_info!("Skipping SMP CPU test on single-CPU system\n");
+        return Ok(Pass);
+    }
```

Comment:
I think we can remove this, because if there is only one CPU, the following
```rust
let remote_id = remote.call(closure)?;
```
, will return an error, and skip the test.

This is one of the cases when the test can only partially run and we have to skip (due to lacking of some resource).

Reply:
It'll fail before remote.call `Cpu::from_id` won't accept `1` on a single-CPU machine. That said, this isn't here out of necessity, it's here to make clear that we're _intentionally_ skipping this test. It seems like error'ing out should be treated as equivalent to failure, rather than simply skipping the test?